pkg/virtfs/virtfs.go

```diff
@@ -0,0 +1,72 @@
+package virtfs
+
+import (
+	"os/exec"
+	"path/filepath"
+	"strconv"
+
+	"github.com/lima-vm/sshocker/pkg/ssh"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type VirtFS struct {
+	*ssh.SSHConfig
+	LocalPath            string
+	MountTag             string
+	Host                 string
+	Port                 int
+	RemotePath           string
+	Readonly             bool
+	sshCmd               *exec.Cmd
+	VirtFSAdditionalArgs []string
+}
+
+func (vfs *VirtFS) sshCommand(args ...string) ([]byte, error) {
+	sshBinary := vfs.SSHConfig.Binary()
+	sshArgs := vfs.SSHConfig.Args()
+	sshArgs = append(sshArgs, "-p", strconv.Itoa(vfs.Port), vfs.Host, "--")
+	sshArgs = append(sshArgs, args...)
+	vfs.sshCmd = exec.Command(sshBinary, sshArgs...)
+	logrus.Debugf("executing ssh: %s %v", vfs.sshCmd.Path, vfs.sshCmd.Args)
+	return vfs.sshCmd.CombinedOutput()
+}
+
+func (vfs *VirtFS) Prepare() error {
+	if !filepath.IsAbs(vfs.RemotePath) {
+		return errors.Errorf("unexpected relative path: %q", vfs.RemotePath)
+	}
+	out, err := vfs.sshCommand("sudo", "mkdir", "-p", vfs.RemotePath)
+	if err != nil {
+		return errors.Wrapf(err, "failed to mkdir %q (remote): %q", vfs.RemotePath, string(out))
+	}
+	return nil
+}
+
+func (vfs *VirtFS) Start() error {
+	if !filepath.IsAbs(vfs.RemotePath) {
+		return errors.Errorf("unexpected relative path: %q", vfs.RemotePath)
+	}
+	sshArgs := []string{"sudo", "mount", "-t", "9p", "-o", "trans=virtio", vfs.MountTag, vfs.RemotePath}
```

Comment:
Shouldn't this be specified in the cloud-init ?

Reply:
Right now it is done with sshocker "libraries".

Reply:
https://cloudinit.readthedocs.io/en/latest/topics/examples.html#adjust-mount-points-mounted